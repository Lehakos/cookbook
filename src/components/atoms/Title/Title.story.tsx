import React, { Fragment } from 'react';
import { storiesOf } from '@storybook/react';

import Title from './Title';

const stories = storiesOf('Atoms|Title', module);

stories
  .add('Default', () => <Title>This is a title!</Title>)
  .add('Show margin', () => (
    <Fragment>
      <Title>This is a title!</Title>
      <div>Some text</div>
    </Fragment>
  ));
