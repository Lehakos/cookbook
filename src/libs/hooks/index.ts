export { useControlledProp } from './use-controlled-prop';
export { useElementSize } from './use-element-size';
