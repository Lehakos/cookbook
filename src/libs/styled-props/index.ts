export { typography, TypographyProps } from './typography';
export { layout, LayoutProps } from './layout';
export { flexbox, FlexboxProps } from './flexbox';
