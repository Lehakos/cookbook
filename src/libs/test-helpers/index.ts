export { render } from './render';
export { mockElementWithBoundingRect } from './mock-element-with-bounding-rect';
