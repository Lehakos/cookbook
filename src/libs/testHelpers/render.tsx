import React, { ReactElement } from 'react';
import { render as baseRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from 'shared/theme';

export const render = (el: ReactElement) =>
  baseRender(<ThemeProvider theme={theme}>{el}</ThemeProvider>);
