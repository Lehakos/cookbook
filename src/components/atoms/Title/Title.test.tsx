import React from 'react';

import { render } from 'libs/testHelpers';

import Title from './Title';

describe('<Title />', () => {
  it('renders properly', () => {
    const { container } = render(<Title />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
