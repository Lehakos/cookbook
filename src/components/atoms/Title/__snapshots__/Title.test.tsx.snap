// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> правильно отображается 1`] = `
.c0 {
  margin: 0;
  margin-bottom: 0.75rem;
  font-size: 1.5rem;
  color: #F47983;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05rem;
  -moz-letter-spacing: 0.05rem;
  -ms-letter-spacing: 0.05rem;
  letter-spacing: 0.05rem;
}

<h1
  class="c0"
/>
`;
